app/models/manageiq/providers/google/cloud_manager/vm.rb

```diff
@@ -1,6 +1,23 @@
 class ManageIQ::Providers::Google::CloudManager::Vm < ManageIQ::Providers::CloudManager::Vm
   include_concern 'Operations'
 
+  virtual_column :preemptible?, :type => :boolean, :uses => :advanced_settings
+
+  def preemptible?
+    preempt_setting = advanced_settings.detect { |v| v.name == "preemptible?" }
+
+    # It's possible that we got a nil back; this would occur if the VM hasn't
+    # been refreshed since this change has been added. If that's the case, log
+    # an error and return false for now.
+    if preempt_setting.nil?
+      _log.warn("Unable to find 'preemptible?' value for vm; this failure is"\
+                " likely temporary and will resolve upon a refresh")
+      return false
+    end
```

Comment:
Since a refresh generally occurs relatively quickly, I wonder if this warning is needed at all.  You definitely need a nil check, but it might be simple enough to just say `preempt_setting.try(:value) == "true"`.